app/services/chatbot/chatbot-betting.ts

```diff
@@ -0,0 +1,321 @@
+    return this.chatbotBaseApiService
+      .api('POST', 'settings/betting', data)
+      .then((response: IChatbotAPIPostResponse) => {
+        if (response.success === true) {
+          this.fetchPreferences();
+          this.chatbotCommonService.closeChildWindow();
+        }
+      });
+  }
+
+  addProfile(data: IBettingProfile) {
+    return this.chatbotBaseApiService.api('POST', 'betting/profile', data).then(() => {
+      this.fetchPreferences();
+      this.chatbotCommonService.closeChildWindow();
+    });
+  }
+
+  updateProfile(data: IBettingProfile) {
+    return this.chatbotBaseApiService
+      .api('PUT', `betting/profile/${data.id}`, data)
+      .then((response: IChatbotAPIPutResponse) => {
+        if (response) {
+          this.fetchPreferences();
+          this.chatbotCommonService.closeChildWindow();
+        }
+      });
+  }
+
+  start(data: IBettingProfile) {
+    return this.chatbotBaseApiService.api('POST', `betting/start/${data.id}`, {}).then(() => {
+      this.fetchActive();
+      this.chatbotCommonService.closeChildWindow();
+    });
+  }
+
+  open() {
+    return this.chatbotBaseApiService.api('PUT', 'betting/active/open', {});
+  }
+
+  close() {
+    return this.chatbotBaseApiService.api('PUT', 'betting/active/close', {});
+  }
+
+  cancel() {
+    return this.chatbotBaseApiService.api('PUT', 'betting/active/cancel', {});
+  }
+
+  complete() {
+    return this.chatbotBaseApiService.api('PUT', 'betting/active/complete', {});
+  }
+
+  pickWinner(option: string) {
+    return this.chatbotBaseApiService.api('POST', 'betting/active/pick', {
+      option,
+    });
+  }
+
+  //
+  // Delete
+  //
+  deleteProfile(data: IBettingProfile) {
+    return this.chatbotBaseApiService
+      .api('DELETE', `betting/profile/${data.id}`, {})
+      .then((response: IChatbotAPIDeleteResponse) => {
+        if (response.success === true) {
+          this.fetchPreferences();
+          this.chatbotCommonService.closeChildWindow();
+        }
+      });
+  }
+
+  //
+  // Reset
+  //
+  resetSettings() {
+    return this.chatbotBaseApiService
+      .resetSettings('betting')
+      .then((response: IBettingPreferencesResponse) => {
+        this.UPDATE_BETTING_PREFERENCES(response);
+        return Promise.resolve(response);
+      });
+  }
+
+  //
+  // Views
+  //
+  changeView(view: string) {
+    this.UPDATE_BETTING_VIEW(view);
+  }
+
+  //
+  //  Timer
+  //
+  @mutation()
+  private UPDATE_TIMER() {
+    const activePoll = this.state.activeBettingResponse;
+    const containsSettings =
+      activePoll.settings !== undefined && activePoll.settings.timer !== undefined;
+
+    if (!containsSettings) {
+      return;
+    }
+
+    const containsTimer = activePoll.settings.timer.enabled;
+    const startedTimer = activePoll.settings.timer.started_at !== undefined;
+
+    if (containsSettings && containsTimer && startedTimer) {
+      const timeElapsed = Date.now() - activePoll.settings.timer.started_at;
```

Comment:
Can't use `Date.now()` here.
Mutation must be a pure function that relies only on arguments and state. This method will be executed in the main and child window and `Date.now()` will be different.
To fix that, just pass a `Date.now()` as an argument for this mutation

Reply:
I need the Date.now() at that point in time since I am calculating the time that has elapsed since the timer has started, it's used for a countdown / count up timer.

Reply:
Good catch.  I missed this.